### DISCUSSION TITLE
Failure building the zephyr\tests\subsys\cpp\libcxx project

### DISCUSSION THREAD

**Discussion by awe-devasc** [zephyrproject-rtos/zephyr#38797] [Q&A] [answered]
Not finding the STL include files ... this is the build error message:
/zephyr/tests/subsys/cpp/libcxx/src/main.cpp:7:10: fatal error: array: No such file or directory
    7 | #include <array>
      |          ^~~~~~~
compilation terminated.

ARM gnu eabi installed and the STL files are there ...  
in the directory ... C:\gnu_arm_embedded\arm-none-eabi\include\c++\10.3.1 
Any advice as to how to get this test working would be much appreciated.

**Comment by stephanosio** [comment, marked as answer]
It looks like you are trying to build `tests/subsys/cpp/libcxx` manually (i.e. not via twister).

https://github.com/zephyrproject-rtos/zephyr/blob/611284021cef3bb014437d5f2bf966d85320cc8e/tests/subsys/cpp/libcxx/testcase.yaml#L5-L13

In that case, you need to make sure that the correct configs are set:
`-DCONFIG_NEWLIB_LIBC=y`
`-DCONFIG_NEWLIB_LIBC_NANO=n `
`-DCONFIG_EXCEPTIONS=y`

**Reply by awe-devasc** [reply]
Thank you this all makes sense ... and it worked ... 
I have not yet explored twister ...
What would be the correct procedure for using twister with this particular example ?
Thanks for all your help and advice
Andrew

**Reply by stephanosio** [reply]
> What would be the correct procedure for using twister with this particular example ?

You can try the following from the top of the Zephyr source tree:
```
scripts/twister -v -N -p [BOARD NAME] -T tests/subsys/cpp/libcxx
```

e.g. with `qemu_cortex_m3`:
```
scripts/twister -v -N -p qemu_cortex_m3 -T tests/subsys/cpp/libcxx
```
